src/frontends/lean/brackets.cpp

```diff
@@ -45,16 +45,24 @@ static expr mk_singleton(parser & p, pos_info const & pos, expr const & e) {
 /* Parse rest of the insertable '{' expr ... */
 static expr parse_fin_set(parser & p, pos_info const & pos, expr const & e) {
     lean_assert(p.curr_is_token(get_comma_tk()) || p.curr_is_token(get_rcurly_tk()));
-    expr r = mk_singleton(p, pos, e);
+    buffer<pair<pos_info, expr>> stack;
+    stack.emplace_back(pos, e);
     while (p.curr_is_token(get_comma_tk())) {
         auto ins_pos = p.pos();
         p.next();
         if (p.curr_is_token(get_rcurly_tk())) break;
         expr e2 = p.parse_expr();
-        expr insert = p.save_pos(mk_constant(get_has_insert_insert_name()), ins_pos);
-        r = p.rec_save_pos(mk_app(insert, e2, r), ins_pos);
+        stack.emplace_back(ins_pos, e2);
     }
     p.check_token_next(get_rcurly_tk(), "invalid explicit finite collection, '}' expected");
+    unsigned i = stack.size() - 1;
+    auto [pos, e2] = stack[i];
```

Comment:
Lean still uses C++11, so this won't work.

Reply:
What's the plan here: rewrite or switch to c++17?

Reply:
While I would be fine with switching to C++17, my plan was to rewrite when I get the time to come back to this.

Reply:
With `g++ -std=c++17` I get the following error:
```
<...>/lean3/src/frontends/lean/brackets.cpp: In function ‘lean::expr lean::parse_fin_set(lean::parser&, const pos_info&, const lean::expr&)’:
<...>/lean3/src/frontends/lean/brackets.cpp:59:18: error: declaration of ‘auto pos’ shadows a parameter
   59 |     auto [pos, e2] = stack[i];
      |                  ^
make[2]: *** [frontends/lean/CMakeFiles/lean_frontend.dir/build.make:427: frontends/lean/CMakeFiles/lean_frontend.dir/brackets.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:2226: frontends/lean/CMakeFiles/lean_frontend.dir/all] Error 2
make: *** [Makefile:163: all] Error 2
```

Reply:
Fixed compile. Feel free to revert/rewrite.